lib/fizzy/parser_expr.cpp

```diff
@@ -148,6 +150,12 @@ parser_result<Code> parse_expr(
     // The function's implicit block.
     control_stack.emplace(Instr::block, function_arity, 0);
 
+    // This is needed for aiding validation.
+    assert((uint64_t{local_count} + module.typesec[func_type_idx].inputs.size()) <=
```

Comment:
Would be nice to declare `const auto& func_type = module.typesec[func_type_idx];`
(you would need to rename a variable in `call` and `call_indirect` cases)

Reply:
Added.